src/installer/managed/Microsoft.NET.HostModel/MachO/CodeSignature.cs

```diff
@@ -0,0 +1,155 @@
+            || cs._embeddedSignature.Header.Requirements.Slot != CodeDirectorySpecialSlot.Requirements
+            || cs._embeddedSignature.Header.CmsWrapper.Slot != CodeDirectorySpecialSlot.CmsWrapper)
+        {
+            cs._unrecognizedFormat = true;
+            return cs;
+        }
+        cs._codeDirectory.FileOffset = fileOffset + cs._embeddedSignature.Header.CodeDirectory.Offset;
+        file.Read(cs._codeDirectory.FileOffset, out cs._codeDirectory.Header);
+        if (cs._codeDirectory.Header.Version != CodeDirectoryVersion.HighestVersion
+            || cs._codeDirectory.Header.HashType != HashType.SHA256)
+        {
+            cs._unrecognizedFormat = true;
+            return cs;
+        }
+
+        cs._identifierPtr.FileOffset = cs._codeDirectory.FileOffset + cs._codeDirectory.Header.IdentifierOffset;
+        cs._cdHashes.FileOffset = cs._codeDirectory.FileOffset + cs._codeDirectory.Header.HashesOffset - MachObjectFile.SpecialSlotCount * MachObjectFile.DefaultHashSize;;
+
+        cs._identifierPtr.Identifier = new byte[cs._cdHashes.FileOffset - cs._identifierPtr.FileOffset];
+        file.ReadArray(cs._identifierPtr.FileOffset, cs._identifierPtr.Identifier, 0, cs._identifierPtr.Identifier.Length);
+
+        cs._cdHashes.Blob = new byte[(MachObjectFile.SpecialSlotCount + cs._codeDirectory.Header.CodeSlotCount) * MachObjectFile.DefaultHashSize];
+        file.ReadArray(cs._cdHashes.FileOffset, cs._cdHashes.Blob, 0, cs._cdHashes.Blob.Length);
+
+        cs._requirementsBlob.FileOffset = fileOffset + cs._embeddedSignature.Header.Requirements.Offset;
+        file.Read(cs._requirementsBlob.FileOffset, out cs._requirementsBlob.Header);
+        if (!cs._requirementsBlob.Header.Equals(RequirementsBlob.Empty))
+        {
+            cs._unrecognizedFormat = true;
+            return cs;
+        }
+
+        cs._cmsWrapperBlob.FileOffset = fileOffset + cs._embeddedSignature.Header.CmsWrapper.Offset;
+        file.Read(cs._cmsWrapperBlob.FileOffset, out cs._cmsWrapperBlob.Header);
+        if (!cs._cmsWrapperBlob.Header.Equals(CmsWrapperBlob.Empty))
+        {
+            cs._unrecognizedFormat = true;
+            return cs;
+        }
+        return cs;
+    }
+
+    internal void WriteToFile(MemoryMappedViewAccessor file)
+    {
+        file.Write(_embeddedSignature.FileOffset, ref _embeddedSignature.Header);
+        file.Write(_codeDirectory.FileOffset, ref _codeDirectory.Header);
+        file.WriteArray(_identifierPtr.FileOffset, _identifierPtr.Identifier, 0, _identifierPtr.Identifier.Length);
+        file.WriteArray(_cdHashes.FileOffset, _cdHashes.Blob, 0, _cdHashes.Blob.Length);
+        file.Write(_requirementsBlob.FileOffset, ref _requirementsBlob.Header);
+        file.Write(_cmsWrapperBlob.FileOffset, ref _cmsWrapperBlob.Header);
+    }
+
+    internal void WriteToStream(FileStream file)
+    {
+        byte[] arr = new byte[(int)Size];
+        Span<byte> buffer = arr;
```

Comment:
`stackalloc` since it's a constant size?

Reply:
This is netstandard2.0 so most of the new stuff doesn't work. In this case, byte[] arr is needed for line 152 (File.Write).